flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/common/CommonPhysicalSink.scala

```diff
@@ -99,11 +105,33 @@ class CommonPhysicalSink (
 
         val operator = new SinkOperator(env.clean(sinkFunction), rowtimeFieldIndex, enforcer)
 
+        val inputParallelism = inputTransformation.getParallelism
+        val taskParallelism = env.getParallelism
+        val parallelism = if (runtimeProvider.isInstanceOf[ParallelismProvider]) runtimeProvider.asInstanceOf[ParallelismProvider].getParallelism.orElse(inputParallelism).intValue()
```

Comment:
Remove `isInstanceOf`, it must be

Reply:
My concern is that once a NEW provider without `parallelismProvider` is used, exception will be thrown unless we do not check the type

Reply:
Add a scala `assert` is enough.